spring-xd-module/src/main/java/org/springframework/xd/module/core/ModuleFactory.java

```diff
@@ -135,6 +144,18 @@ else if (JavaConfiguredModule.basePackages(moduleDefinition, moduleClassLoader).
 		return null;
 	}
 
+	private Module createSparkModule(ModuleDescriptor moduleDescriptor, ModuleOptions moduleOptions,
```

Comment:
Agreed (for now), but eventually we should refactor the ModuleFactory to support a contribution model (also relevant for the comments above about dependencies).